### PR TITLE
add demographic filters for data

### LINKED ISSUE #118: More filters for data
E.g. by demographic or criticality or mode of response (paper/ online/ SMS)

### PR DESCRIPTION
**Pull request by asegun-cod**
close #118

### CODE REVIEW

File: R/mod_demographics_selection.R [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+    output$age_UI <- renderUI({
+      isolate(
+        choices <- filter_data()$unique_data %>%
+          dplyr::arrange(age) %>%
+          dplyr::distinct(age, .keep_all = TRUE) %>%
+          dplyr::filter(!is.na(age))
+      )
+
+      choices <- factor(choices$age,
```
Comment (ChrisBeeley) [inline R/mod_demographics_selection.R:58]: It's a minor point but overwriting a dataframe (`choices`) with a vector seems a bit counter intuitive. Maybe better to have `choices <- factor(choices_df$age...)`?

File: R/mod_demographics_selection.R [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        selected = NULL, multiple = TRUE
+      )
+    })
+
+    output$gender_UI <- renderUI({
+      isolate(
+        choices <- filter_data()$unique_data %>%
+          dplyr::arrange(gender) %>%
+          dplyr::distinct(gender) %>%
+          dplyr::filter(!gender == "")
```
Comment (ChrisBeeley) [inline R/mod_demographics_selection.R:74]: Again a minor point but I would prefer `dplyr::filter(gender != "")`

File: R/mod_fft.R [hunk 3]
```diff
@@ -43,7 +43,13 @@ mod_fft_server <- function(id, filter_data){
       
     graph_data <- reactive({
       
-      split_data_spc(filter_data()$unique_data, variable = "fft", chunks = 'monthly')
+      tryCatch({
```
Comment (ChrisBeeley) [inline R/mod_fft.R:46]: Why is this in a tryCatch block? What error does it throw and why?

**Review by ChrisBeeley — COMMENTED** [review]
Some minor comments, merge whenever you like

File: R/mod_demographics_selection.R (hunk 2 above)
Comment (asegun-cod) [reply R/mod_demographics_selection.R:74]: honestly, i never knew this word. i prefer it just that I haven't tried it. i will do now

File: R/mod_fft.R (hunk 3 above)
Comment (asegun-cod) [reply R/mod_fft.R:46]: I guess I was meant to remove this. I added it when trying some things

File: R/mod_demographics_selection.R [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+      )
+
+      selectInput(session$ns("select_gender"),
+        label = "Select gender (defaults to all)",
+        choices = na.omit(choices) %>% dplyr::pull(gender),
```
[the remark this answers is not in the document]
Comment (asegun-cod) [reply R/mod_demographics_selection.R:79]: i used  styler:::style_active_file() to tidy the code. it wasn't picked so I will have to manually resolve that

File: R/mod_fft.R (hunk 3 above)
Comment (asegun-cod) [reply R/mod_fft.R:46]: Sorry, just figured this out. the `trycatch()` was in the wrong part. the error arises when the selected date range is not enough to do `date = as.Date(cut(date, "month"))` in `split_data_spc()`

**Pull request merged**